### PR TITLE
[process-agent] Remove host info from AgentConfig

### PR DESCRIPTION
**Pull request by xornivore**
### What does this PR do?

- Remove hostname and container host type from `AgentConfig`
- Refactor check initialization
- Stop passing `AgentConfig` to `Run` methods for checks.

The only two fields remaining in `AgentConfig` are `MaxConnsPerMessage` and `SystemProbeAddress` and both can be read directly from the system probe config.

### Motivation

Prep for removing pkg/process/config.

### Describe how to test/QA your changes

- Validate common check functionality and agent initialization works correctly.
- Validate status still shows the hostname correctly.

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature. 
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

File: cmd/process-agent/collector_api_test.go [hunk 1]
```diff
@@ -528,7 +528,7 @@ func runCollectorTest(t *testing.T, check checks.Check, cfg *config.AgentConfig,
 	runCollectorTestWithAPIKeys(t, check, cfg, epConfig, []string{"apiKey"}, []string{"orchestratorApiKey"}, mockConfig, init, tc)
 }
 
-func runCollectorTestWithAPIKeys(t *testing.T, check checks.Check, cfg *config.AgentConfig, epConfig *endpointConfig, apiKeys, orchAPIKeys []string, mockConfig ddconfig.Config, init func(c *Collector), tc func(c *Collector, ep *mockEndpoint)) {
+func runCollectorTestWithAPIKeys(t *testing.T, check checks.Check, _ *config.AgentConfig, epConfig *endpointConfig, apiKeys, orchAPIKeys []string, mockConfig ddconfig.Config, init func(c *Collector), tc func(c *Collector, ep *mockEndpoint)) {
```
Comment (just-chillin) [inline cmd/process-agent/collector_api_test.go:531]: Since `cfg` is now unused, can we remove it from the function parameters?

**Review by just-chillin — APPROVED** [review]
Looks great!

File: cmd/process-agent/collector_api_test.go (hunk 1 above)
Comment (just-chillin) [reply cmd/process-agent/collector_api_test.go:531]: Ah, never mind it's removed here https://github.com/DataDog/datadog-agent/pull/14889/files#diff-cb0e545e5c9aba9e9985490d4092d686b4444ee971115d6f9ab2624fcf3968baR525

File: cmd/process-agent/info_test.go [hunk 2]
```diff
@@ -75,7 +74,7 @@ func (h *testServerHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
...
...56610,99514,430100,155465,687228,2882711,6731693,65804,1624821,159652,285577,60549,157904,5916134,412288,132597,908879,272218,180757,732263,169132,745769,306887,7252246,4394707,951611,82203,213042,117441,112781,112222,300525,1211909,145623,113760,164115,225506,96600,953395,1874155,204582,115017,187433,130060,156279,312904,97728,2302220,232036,124466,632575,121143,149310,200161,248362,153779,119662,494180,534981,316531,185806,152670,211013,132662,2033315,253769,850340,299891,140935,120561,170304,87502,712511,758267,102834,472697,1307030,24962796,254714,2628369,266971,1318833,108503,131614,120129,222602,1789785,129309,746252,19088113,823822,253070,163050,183856,123350,126196,226394,97382,152971,1404152,196053,1299475,136466,185384,20561432,12951037,1264979,5630592,2187745,254259,1811039,164578,144580,244778,207351,2194256,12554634,16700526,5939874,18788363,9003636,379686,92515,771011,1034056,795588,842805,313889,553591],"PauseTotalNs":833945600,"StackInuse":983040,"StackSys":983040,"Sys":11114744,"TotalAlloc":1301199968},"pid":485,"process_count":84,"queue_size":0,"splitter":{},"uptime":3464,"version":{"Major": 0,"Minor": 99,"Patch": 0},"log_file":"/var/log/datadog/process-agent.log"
```
Comment (kkhor-datadog) [inline cmd/process-agent/info_test.go:77]: I think the changes in this file might cause merge conflicts with the changes to `info_test.go` in https://github.com/DataDog/datadog-agent/commit/b806b3be91f7b196df8006886db541ea803a2279. Not sure if git will automatically handle it correctly, so just something to be aware of during the merge or rebase

File: cmd/process-agent/main_common.go [hunk 3]
```diff
@@ -230,7 +225,7 @@ func runAgent(globalParams *command.GlobalParams, exit chan struct{}) {
 	if globalParams.Info {
 		// using the debug port to get info to work
 		url := fmt.Sprintf("http://localhost:%d/debug/vars", expVarPort)
-		if err := Info(os.Stdout, cfg, url); err != nil {
+		if err := Info(os.Stdout, url); err != nil {
```
Comment (mbotarro) [inline cmd/process-agent/main_common.go:228]: Is the `--info` command still showing the hostname now that `initInfo` is called after Info()?

**Review by mbotarro — APPROVED** [review]
Looks great! 🔥

File: cmd/process-agent/info_test.go (hunk 2 above)
Comment (xornivore) [reply cmd/process-agent/info_test.go:77]: Thank you, looks like it's not flagging it as a merge conflict with main, so I am hopeful 😄

File: cmd/process-agent/main_common.go (hunk 3 above)
Comment (xornivore) [reply cmd/process-agent/main_common.go:228]: Good question, let me verify again. Thank you.

**Review by JLineaweaver — APPROVED** [review]
LGTM. Nothing really stands out that looks concerning. 👍

File: cmd/process-agent/main_common.go (hunk 3 above)
Comment (xornivore) [reply cmd/process-agent/main_common.go:241]: You were absolutely right @mbotarro, fixed it in [9813017](https://github.com/DataDog/datadog-agent/pull/14885/commits/9813017c61c7ca5e74ff083c454a2b72ec9cdbdd)

Comment (mbotarro) [reply cmd/process-agent/main_common.go:241]: Thanks for the fix!!

**Review by mbotarro — APPROVED** [review]

**Review by hmahmood — APPROVED** [review]

**Pull request merged**